### PR TITLE
3782 - Add doc lint github action

### PR DESCRIPTION
**Pull request by diana-borbe-consensys**
This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,7 +1,35 @@
+## Description
+<!-- Provide a brief description of the changes in this PR -->
+
+## Related Issue
+
 This PR implements issue(s) #
```
Comment (jpnovais) [inline .github/pull_request_template.md]:
```suggestion
This PR addresses/relates to issue(s) #
```
I suggest that as PR may not necessarily implement the issue (many times a feature is implemented in multiple PRs)

File: .github/workflows/doc-lint.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  #       run: |
+  #         echo "${{ env.NODE_VERSION }}" > .nvmrc
+  #     - name: Lint
+  #       uses: Consensys/docs-gha/lint@main
+  #       with:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  linkCheck:
+    name: Link Checking
+    runs-on: [self-hosted, ubuntu-22.04, X64, small]
```
Comment (jpnovais) [inline .github/workflows/doc-lint.yaml:33]: Does this `ubuntu-22.04` work? That good :) 

@bradbown FYI

Comment (diana-borbe-consensys) [reply .github/workflows/doc-lint.yaml:33]: Job completed successfully.

**Review by jpnovais — APPROVED** [review]

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
-* [ ] I have successfully ran tests, style checker and build against my new changes locally.
-* [ ] I have informed the team of any breaking changes if there are any.
\ No newline at end of file
+This PR addresses/relates to issue(s) #
+<!-- If this PR addresses an existing issue, please link to it here -->
+<!-- e.g., Fixes #123 -->
+
+## Type of Change
+<!-- Mark the appropriate option with an "x" -->
+- [ ] Bug fix (non-breaking change which fixes an issue)
```
Comment (thedarkjester) [inline .github/pull_request_template.md:12]: Do we need an additional section for `hotfix` or do you think `Bug fix` already covers it?

Comment (diana-borbe-consensys) [reply .github/pull_request_template.md:12]: Can include both in same category. Do you prefer to keep them separate ?

**Pull request closed without merging**